### PR TITLE
{Storage} Build azure stack context 

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

To separate Azure Cloud and Azure Stack in storage module, we need to build different context with separated file system:
- commands.py : command table
- _params.py: argument context
- _client_factory.py: client factory
- _transformers.py: transform operations
- _validators.py: validate functions

Note：
- _formats.py is used for table transformer operations
- _transformer.py is used for transform operations

**Testing Guide**  

`azdev test storage`
`azdev test storage --profile 2019-03-01-hybrid`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli/azure/cli/command_modules/storage/_transformers_azure_stack.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    for i, item in enumerate(new_result):
+        new_result[i] = transform_fs_public_access_output(item)
+    return new_result
+
+
+def transform_metadata(result):
+    return result.metadata
+
+
+# ------------------Track2 Support-----------------------
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_transformers_azure_stack.py:175]: We should only put the transform for Track 1 here, right?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_transformers_azure_stack.py:175]: yes

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/storage/_transformers_azure_stack.py:175]: So why not remove this redundant code here?

**Pull request merged**